python/cudf/cudf/core/series.py

```diff
@@ -2597,6 +2597,10 @@ def is_monotonic_decreasing(self):
     def __cuda_array_interface__(self):
         return self._column.__cuda_array_interface__
 
+    def repeat(self, repeats):
```

Comment:
We also need to hook this function up to the Index class as well.

Reply:
Should probably move it down to the `column` level and then have both Series and index call into the column.

Reply:
You asked to handle the index https://github.com/rapidsai/cudf/pull/2802#discussion_r325165727. How do I go about doing that in `column` level?

Reply:
The index is backed by a column, accessed via `index._values` unless it's a MultiIndex which is then backed by a DataFrame via `multiindex._source_data`.

Given both Series and Index are backed by columns, it would make sense to define the `repeat` function at the column level and then call that from the Series/Index classes.

Reply:
Would this suffice?
```python
class Index:
  def repeat(self, repeats):
    return self.to_series().repeat(repeats).as_index()
```
Bear in mind that the series method handles its own index as well.

Reply:
That's a super anti-pattern of tech debt we should eliminate. It would be much cleaner if we could:
```
class Index:
    def repeat(self, repeats):
        return as_index(self._values.repeat(repeats))
```

Then Series could be:
```
class Series:
    def repeat(self, repeats):
        data = self._column.repeat(repeats)
        new_index = self.index.repeat(repeats)
        return cls(data, index=new_index, name=self.name)
```